### PR TITLE
19414 - Account Management - add search for closed/historical accounts

### PR DESCRIPTION
**Pull request by sameer0422**
*Issue #:*
https://github.com/bcgov/entity/issues/19414

*Description of changes:* Account Management - add search for closed/historical accounts

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2850-k4dxwtt4.web.app

File: auth-web/src/components/auth/common/AccountInActiveAlert.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator'
+import CommonUtils from '@/util/common-util'
+import { Organization } from '@/models/Organization'
+import { State } from 'pinia-class'
+import { useOrgStore } from '@/stores/org'
+
+@Component
```
Comment (seeker25) [inline auth-web/src/components/auth/common/AccountInActiveAlert.vue:39]: Please rewrite using `composition-api`

File: auth-web/src/components/auth/common/AccountInActiveAlert.vue [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+export default class AccountInActiveAlert extends Vue {
+  @State(useOrgStore) private currentOrganization!: Organization
+  private formatDate = CommonUtils.formatDisplayDate
+
+  get deactivatedDate () {
+    return (this.currentOrganization?.modified)
+      ? this.formatDate(new Date(this.currentOrganization.modified)) : ''
```
Comment (seeker25) [inline auth-web/src/components/auth/common/AccountInActiveAlert.vue:46]: Please use `moment` if possible `new Date(<string>)` has issues

File: auth-web/src/components/auth/common/AccountInActiveAlert.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+  .account-alert-inner {
+    display: flex;
```
Comment (seeker25) [inline auth-web/src/components/auth/common/AccountInActiveAlert.vue:53]: there's a Vuetify class for this

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue [hunk 4]
```diff
@@ -96,6 +96,14 @@
           </v-badge>
         </v-tab>
       </template>
+
+      <v-tab
+        v-if="canViewAccounts"
+        data-test="inactive-tab"
+        :to="pagesEnum.STAFF_DASHBOARD_InACTIVE"
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue:103]: Typo? InACTIVE?

should be INACTIVE?

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue [hunk 5]
```diff
@@ -96,6 +96,14 @@
...
+      >
+        InActive
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue:105]: `Inactive`

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue [hunk 6]
```diff
@@ -129,7 +138,8 @@ enum TAB_CODE {
     PendingReview = 'pending-review-tab',
     Rejected = 'rejected-tab',
     Invitations = 'invitations-tab',
-    Suspended = 'suspended-tab'
+    Suspended = 'suspended-tab',
+    InActive = 'inactive-tab'
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue:142]: `Inactive`

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue [hunk 7]
```diff
@@ -138,6 +148,7 @@ enum TAB_CODE {
     StaffPendingAccountsTable,
     StaffRejectedAccountsTable,
     StaffPendingAccountInvitationsTable,
+    StaffInActiveAccountsTable,
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue:151]: `Inactive`

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue [hunk 8]
```diff
@@ -198,6 +209,11 @@ export default class StaffAccountManagement extends Vue {
       id: 4,
       tabName: 'Suspended',
       code: TAB_CODE.Suspended
+    },
+    {
+      id: 5,
+      tabName: 'InActive',
+      code: TAB_CODE.InActive
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue:216]: `Inactive`

File: auth-web/src/components/auth/staff/account-management/StaffInActiveAccountsTable.vue [hunk 9]
```diff
@@ -0,0 +1,539 @@
...
+import debounce from '@/util/debounce'
+import { useOrgStore } from '@/stores/org'
+import { useStaffStore } from '@/stores/staff'
+
+@Component({
+  components: {
+    SearchFilterInput
+  }
+})
+export default class StaffInActiveAccountsTable extends Mixins(PaginationMixin) {
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffInActiveAccountsTable.vue:171]: Please rewrite using `composition-api`

File: auth-web/src/util/constants.ts [hunk 10]
```diff
@@ -90,6 +91,7 @@ export enum Pages {
     STAFF_DASHBOARD_REJECTED = '/staff/dashboard/rejected',
     STAFF_DASHBOARD_INVITATIONS = '/staff/dashboard/invitations',
     STAFF_DASHBOARD_SUSPENDED = '/staff/dashboard/suspended',
+    STAFF_DASHBOARD_InACTIVE = '/staff/dashboard/inactive',
```
Comment (seeker25) [inline auth-web/src/util/constants.ts:94]: `InACTIVE` -> `INACTIVE`

File: auth-web/src/views/auth/AccountSettings.vue [hunk 11]
```diff
@@ -97,12 +97,19 @@
...
       </p>
     </div>
+
     <!-- Suspend Account Banner-->
     <AccountSuspendAlert
       v-if="showAccountFreezeBanner"
       class="account-alert mb-0"
     />
 
+    <!-- InActive Account Banner-->
```
Comment (seeker25) [inline auth-web/src/views/auth/AccountSettings.vue:107]: `Inactive`

File: auth-web/src/components/auth/staff/account-management/StaffInActiveAccountsTable.vue [hunk 12]
```diff
@@ -0,0 +1,539 @@
...
+    return `${tag}-${index}`
+  }
+
+  // Used to go from 'Basic' -> accessType: AccessType.REGULAR, orgType: Account.BASIC
+  protected getOrgAndAccessTypeFromAccountType (accountType: string): object {
+    return this.accountTypeMap[accountType]
+  }
+
+  // Used to go from OrgType -> OrgAccountTypes
+  protected getAccountTypeFromOrgAndAccessType (org:Organization): any {
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffInActiveAccountsTable.vue:349]: Is this code duplicated somewhere? if it is.. probably best to refactor it?

**Review by seeker25 — CHANGES_REQUESTED** [review]
Looks good so far, but should be in composition-api

**Comment by seeker25** [comment]
use v-sanitize:

image

**Comment by seeker25** [comment]
Refactor, so there is less duplication:

image

File: auth-web/src/components/auth/staff/account-management/StaffInActiveAccountsTable.vue [hunk 13]
```diff
@@ -0,0 +1,539 @@
+<template>
+  <div>
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/account-management/StaffInActiveAccountsTable.vue:2]: Unit test for this?

File: auth-web/src/components/auth/common/AccountInActiveAlert.vue (hunk 1 above)
Comment (sameer0422) [reply auth-web/src/components/auth/common/AccountInActiveAlert.vue]: Fixed

File: auth-web/src/components/auth/common/AccountInActiveAlert.vue (hunk 2 above)
Comment (sameer0422) [reply auth-web/src/components/auth/common/AccountInActiveAlert.vue]: Fixed

File: auth-web/src/components/auth/common/AccountInActiveAlert.vue (hunk 3 above)
Comment (sameer0422) [reply auth-web/src/components/auth/common/AccountInActiveAlert.vue]: Fixed

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue (hunk 4 above)
Comment (sameer0422) [reply auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue]: Fixed

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue (hunk 5 above)
Comment (sameer0422) [reply auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue]: Fixed

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue (hunk 6 above)
Comment (sameer0422) [reply auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue]: Fixed

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue (hunk 7 above)
Comment (sameer0422) [reply auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue]: Fixed

File: auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue (hunk 8 above)
Comment (sameer0422) [reply auth-web/src/components/auth/staff/account-management/StaffAccountManagement.vue]: Fixed

File: auth-web/src/util/constants.ts (hunk 10 above)
Comment (sameer0422) [reply auth-web/src/util/constants.ts]: Fixed

File: auth-web/src/views/auth/AccountSettings.vue (hunk 11 above)
Comment (sameer0422) [reply auth-web/src/views/auth/AccountSettings.vue]: Fixed

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2850-k4dxwtt4.web.app

**Comment by seeker25** [comment]
Looks good, maybe 1 suggestion? the deactivated date? or updated_on show that in this screen?

image

File: auth-web/src/views/auth/AccountSettings.vue [hunk 14]
```diff
@@ -454,6 +463,13 @@ export default class AccountSettings extends Mixins(AccountMixin) {
     )
   }
 
+  // show baner for staff user and account inactive
```
Comment (seeker25) [inline auth-web/src/views/auth/AccountSettings.vue:466]: a bit redundant?

**Review by seeker25 — APPROVED** [review]
Great job

**Pull request merged**